cloudformation-template.yaml

```diff
@@ -1718,10 +1718,29 @@ Resources:
       SecretString:
         Ref: SentryEmailPassword
 
+  S3KmsKey:
+    Type: AWS::KMS::Key
+    Properties:
+      KeyPolicy:
+        Version: "2012-10-17"
+        Statement:
```

Comment:
The bucket policy will cause Sentry to error out due to it being unable to decrypt data from S3 once encrypted at rest.

A correct policy would look something along these lines:

```
  S3KmsKey:
    Type: AWS::KMS::Key
    Properties:
      EnableKeyRotation: true
      KeyPolicy:
        Version: "2012-10-17"
        Statement:
          - Effect: Allow
            Action: kms:*
            Resource: "*"
            Principal:
              AWS:
                - Fn::Sub: arn:aws:iam::${AWS::AccountId}:root
          - Effect: Allow
            Principal:
              AWS: "*"
            Action:
              - kms:Encrypt
              - kms:Decrypt
              - kms:ReEncrypt*
              - kms:GenerateDataKey*
              - kms:CreateGrant
              - kms:DescribeKey
            Resource: "*"
            Condition:
              StringEquals:
                kms:CallerAccount:
                  Ref: AWS::AccountId
                kms:ViaService:
                  Fn::Sub: s3.${AWS::Region}.amazonaws.com
```

Notice the 2nd statement to ensure that decryption can be possible.

Reply:
Thanks for the hint!